### DISCUSSION TITLE
Entrando no mercado, saber meu nível e como me vender?

### DISCUSSION THREAD

**Discussion by igorthurow** [frontendbr/forum#968] [Discussão]
Boa noite! Tudo bem?

Eu saí da área de infra ano passado após minha saída de uma empresa de tecnologia chamada Linx.

Já gostava de front, fiz alguns dashs internos na época, mas coisa simples.
Desde agosto do ano passado, tenho estudado bastante, fiz vários cursos de HTML e CSS, digo vários porque passei por teamtreehouse, freecodecamp, school of net, alura... E fui absorvendo um pouco de cada, concluindo etapas, e logo de cara consegui meu primeiro freela. 
Agora estou estudando JS através do curso JS ninja, fantástico! E fiz meu próprio site, tentando alguma visibilidade.

Eu estou com muita vontade de entrar logo para o mercado de trabalho, quero me desenvolver mais rápido, trabalhar com bons profissionais e poder contribuir com algo maior.
Acompanho vários issues de assuntos parecidos aqui no github, mas minha dúvida propriamente dita é se eu realmente já estou apto pra competir pra uma vaga de Junior?
Como me vender? Me descrever? O CV em "papel" ainda é importante mesmo com meu site? Mando e-mails pra todas as vagas me descrevendo? É uma boa pratica enviar o site junto no e-mail? **O que esperar de uma entrevista? Como se preparar?**

Quero aumentar minha visibilidade, e como conclui há poucos dias meu site, o próximo passo é criar um linkedin, eu estou postergando porque preciso de uma boa foto, e quero uma excelente descrição.
Eu estou focado em estudar, estudo bastante, mas o emprego é o foco no momento, as vezes da uma apreensão com a incerteza de conseguir me colocar no mercado. Acredito que vocês já devem ter passado por isso?!

Deixo meu site: http://www.igorthurow.me e o "curriculo" que estou enviando junto nos e-mails por enquanto http://www.igorthurow.me/CV 
(Falta minificar os códigos, é que ainda quero arrumar umas coisinhas, tipo a logo que falta)
Obrigado!

**Comment by Marlysson** [comment]
Eita, portfolio massa man.. Fez ele todo na mão? .. ou usou libs.. frameworks..

**Comment by rduster** [comment]
Tchê, muito legal seu site! Sou de Esteio/RS e estou na mesma jornada, sou formado em Engenharia de Controle e Automação e estou desde setembro do ano passado estudando desenvolvimento web, fiz cursos como freecodecamp, codeacademy, diversos cursos no udemy, no momento estou cursando um bem interessante (em inglês) do Brad Traversy, o nome é Modern JavaScript from the Beginning, recomendo! Vou te enviar um e-mail para trocarmos experiências e nos ajudarmos a encontrar nosso espaço nesse mercado! Abração e sucesso!

**Comment by igorthurow** [comment]
@Marlysson Po, muito obrigado! Cara, sim, fiz na mão... Apesar de ter me baseado em vários sites gringos e brasileiros de vários caras que eu admiro. Foi um desafio! Tive que usar muito inspect e ir pro codepen, github, sabe como é... Meu ultimo freela havia sido em bootstrap 3, então ainda teve essa migraçãozinha. Basicamente Bootstrap 4, JS e Jquery. Um pouco de transform do CSS (ainda não domino, mas consegui ter minha primeira experiência). Sobre técnicas e tecnologia, arrisco dizer que não tem nada muito complexo, porém apliquei alguns conceitos de design flat. Que bom que gostou!!

**Comment by igorthurow** [comment]
@rduster Velho, e eu que sou de Sapucaia? Ahahahaha me manda esse e-mail, vamos marcar um café/cerva essa semana ainda. Aliás, MUITO obrigado!

**Comment by julioxavierr** [comment]
Parabéns pelo site! Gostei muito!
Também estou nessa jornada de conseguir me colocar no mercado e acredito que o essencial é ter vontade de aprender, correr atrás e estudar diariamente. Eu recomendaria que você buscasse agora entrar cada vez mais no mundo JS - frameworks, libraries, ter algumas noções de back-end e etc.

**Comment by wmartins** [comment]
Que legal essa issue @igorthurow, vou tentar responder focando nas tuas perguntas.

> Acompanho vários issues de assuntos parecidos aqui no github, mas minha dúvida propriamente dita é se eu realmente já estou apto pra competir pra uma vaga de Junior?

Normalmente cada empresa/vaga tem a sua própria expectativa para o nível de senioridade do cargo. As vezes um cargo júnior de uma vaga pode ter mais responsabilidades que o mesmo júnior de outra. Entretanto, normalmente (se a empresa é organizada e "responsável"), alguém júnior vai ter o acompanhamento de pessoas mais sênior, tendo um espaço para aprender e se desenvolver. Claro que isso não é regra, vale então conhecer bem as empresas para entender se é uma boa oportunidade para o teu desenvolvimento de carreira.

Particularmente falando, olhando para o teu site como "produto final" (não olhei o código), dá para ver que tu coloca um nível de dedicação bastante bom, além de ver que tu está sempre indo atrás de estudar, se aperfeiçoar e etc. Digo isso só para dizer que, tecnicamente, eu acho que tu tem as skills necessárias para uma vaga Junior.

> Como me vender? Me descrever? O CV em "papel" ainda é importante mesmo com meu site? Mando e-mails pra todas as vagas me descrevendo? É uma boa pratica enviar o site junto no e-mail?

Sobre o CV em "papel", é a forma "padrão" das empresas te conhecerem, então ter um Linkedin e um CV "formatado" é uma forma legal da empresa te conhecer. Tu pode, adicionalmente, incluir um link para o teu site em ambos. Se a empresa não pedir o CV em "papel", tu pode só mandar o site que acho que não tem problema. Acho que tu pode te descrever no CV e no teu site que é suficiente, quem for olhar o teu CV vai te conhecer um pouco melhor.

> O que esperar de uma entrevista? Como se preparar?

Falando das entrevistas, quase todas as empresas que fiz entrevista avaliam o candidato em aspectos técnicos e pessoais, afinal, como tu provavelmente não vai trabalhar sozinho, não basta ser extremamente bom tecnicamente, é também muito importante saber te comunicar, passar uma ideia, discutir, argumentar e etc.

A avaliação técnica normalmente é feita em cima de código, seja algum desafio que a própria empresa te envia (implementar um algoritmo, implementar algum sitezinho, etc.). Porém, também é legal que te avaliem por outros códigos teus, portanto, é legal tu deixar algumas coisas no teu GitHub, visto que isso te expõe um pouco mais para quem está te avaliando. Portanto, coloca lá alguns projetos teus, mesmo que algumas coisas sejam "testes" e etc. Um exemplo legal é disponibilizar o código do teu site, pois aí dá para ver se tu usou algum transpilador (SASS, Babel, etc.), como tu organiza o teu código, como tu documenta o que tu faz e etc.

Se tem teste de código, normalmente é feita uma outra entrevista onde tu explica o código que tu fez, e então a pessoa que está te avaliando vai entender um pouco melhor como tu pensa. As vezes é pedido para que o candidato na hora mexa em alguma coisa do código, fazendo um pair programming com o avaliador. Isso é importante para ter uma ideia de como tu trabalha em equipe. Porém, lembra que isso não é sempre, existem casos que esses processos não acontecem ou acontece em um formato diferente.

Nessa avaliação técnica, quase sempre o avaliador irá tentar explorar as coisas que tu sabe ou que tu está estudando, portanto, tu não precisa se sentir "envergonhado" de não saber algo. Isso é feito para saber qual o teu nível e quais são as coisas que tu vai ter que aprender caso seja contratado. Por exemplo, digamos que a empresa usa React, porém, tu não manja muito, mas tu está disposto a aprender, provavelmente a empresa vai te passar algum treinamento/curso para que o trabalho "flua".

### Dicas

Me estendi um pouco na resposta, mas gostaria de dar algumas outras dicas que podem ser úteis na tua jornada:

- Procure empresas boas e que te permitam crescer. Normalmente trabalhar com gente mais experiente/projetos desafiadores te permite aprender muito. Vale dar uma olhada no Glassdoor para conhecer melhor as empresas.
- Procure vagas que sejam boas **pra ti**. Se a vaga tem tecnologias que tu não gosta/não quer trabalhar, talvez não valha a pena aplicar.
- Aplique e faça entrevistas. Essa é uma boa forma de tu conhecer como as entrevistas em si funcionam, e mesmo que tu não seja contratado, vai ser um aprendizado.
- Se mantenha atualizado e entenda as coisas "como um todo". Por mais que tu queira se especializar em front end, é legal entender um pouco de tudo, ao menos para poder "se virar" em alguma situação.

Espero conseguir te ajudar! :)

**Comment by igorthurow** [comment]
@wmartins Wow! Velho, muito obrigado, que retorno ótimo! Me ajudou a esclarecer várias questões na minha cabeça, e pelo menos dar uma tranquilizada, as vezes parece que nunca seremos bons o suficiente. Vou seguir tuas dicas, vou seguir estudando e me aperfeiçoando e tenho certeza que uma hora vou ser bem recompensando. 

Curioso que quero muito que minha primeira experiência seja numa agência, porque tenho a impressão que é um ambiente mais artístico e "acolhedor", apesar de nunca ter trabalhado em uma. Vou fazer meu linkedin hoje, eu ainda não tenho e acredito que seja de muita importância, afinal todo mundo sempre ressalta muito na hora de conseguir um emprego. 

Novamente: Obrigado! Comecei minha semana com muita motivação!

**Comment by Cleysonlb** [comment]
Falae @igorthurow Beleza?! Cara primeiramente muito legal seu site, da pra ver que vc tem um entendimento legal de front e se preocupa com usabilidade, isso são pontos mega positivos 👍 
Respondendo algumas das suas perguntas.

#### se eu realmente já estou apto pra competir pra uma vaga de Junior?
Claro que sim! Essas vagas existem com o intuito de contratar alguém que tenha uma noção e que tenha vontade de aprender, então vai fundo!!

 #### Como me vender?
Você já esta indo por um ótimo caminho criando seu site e participando de forums.
Tenha segurança em si, mantenha seu linkedin atualizado, tenha presença online e contato com pessoas da área, isso com toda certeza vai te ajudar a se "auto vender".

#### O que esperar de uma entrevista? Como se preparar?

primeiramente mantenha sempre a calma haha, entenda da parte técnica para conseguir conversar/argumentar em uma entrevista, pode ser que algumas empresas peçam que você faça um teste condizente ao nível da vaga.

#### Dica

Sei que as vezes a gente não pode escolher muito... infelizmente temos que pagar as contas, mas de preferencia a empresas com uma boa cultura, empresas que tenham esse perfil de alavancar a carreira de quem esta la dentro, se possivel procure sobre a empresa e sobre quem trabalha lá.

👍 Boa sorte irmão!

**Comment by igorthurow** [comment]
@Cleysonlb Muito obrigado! Só motivação, só ideias pra frente, positiva! Isso só aumenta o combustível e a vontade de crescer mais.

Mandei um CV pra uma excelente empresa do RS, e aparentemente consegui uma entrevista!!! A guria do RH só mandou e-mail dizendo que vai me ligar. Tô tenso hahaahaha, mas vamos nessa. Qualquer coisa já vira experiência. 

Já me desejem sorte!!!!!

**Comment by leandroruel** [comment]
@igorthurow antes de fazer entrevista algumas empresas aplicam algumas provinhas, se prepare para elas antes, estude as tags html 5 (header, main, aside e etc...) e se possível um pouco de js se der tempo, se não conseguir fazer não fique frustrado, acontece.

**Comment by igorthurow** [comment]
@leandroruel Valeu meu velho! Sabe onde posso simular esse tipo de provinha? Acho que domino o assunto, de forma geral, mas pode ser que não, né?! Então é bom estudar e ficar pronto pros "pega-ratão"

**Comment by felipefialho** [comment]
@igorthurow [Criei um repositório com essa intenção](https://github.com/LFeh/frontend-challenges/), lá você pode se testar ;)

**Comment by rduster** [comment]
@LFeh muito obrigado por compartilhar este repositório, venho acompanhando o tópico do Igor pois estou na mesma situação, estes testes vão me ajudar muito a me preparar melhor para as entrevistas! Abraço!

**Comment by leandroruel** [comment]
e todo dia no café da manhã façam um desafio do https://www.codewars.com/

**Comment by igorthurow** [comment]
@LFeh Muito obrigado! FODA! No futuro vou tentar retribuir.

**Comment by igorthurow** [comment]
@leandroruel Perfect. Era justamente o que eu precisava haha valeu!

**Comment by igorthurow** [comment]
E ai galera!!

Caso essa issue seja útil a quem está com questionamentos parecidos, gostaria de informar o desfecho dessa história, que tá na issue [#1016](https://github.com/frontendbr/forum/issues/1016).

Também, preciso expressar a importância de conhecer esse universo que é o frontendbr para minha tomada de decisões e como acelerou minha absorção de conhecimento: a partir de dúvidas em comuns com outros colegas do fórum.

Inicia-se um novo ciclo! Muito obrigado a todos até aqui, um dia espero poder ajudar tanto quanto me ajudaram.
